### PR TITLE
[Data-700] extra params for audio input

### PR DESCRIPTION
**Pull request by bhaney**
Don't merge until https://github.com/viamrobotics/api/pull/93 is merged

This adds "extra" params for the audio input component. 

I'm not exactly sure how useful adding extra is to the audio component - but this is the first pass at doing it.

Since camera is very similar to audio, whatever audio input ends up looking like, the same will happen to camera.

### CODE REVIEW

File: components/audioinput/client.go [hunk 1]
```diff
@@ -45,7 +46,7 @@ func NewClientFromConn(ctx context.Context, conn rpc.ClientConn, name string, lo
 }
 
 func (c *client) Read(ctx context.Context) (wave.Audio, func(), error) {
-	stream, err := c.Stream(ctx)
+	stream, err := c.Stream(ctx, nil)
 	if err != nil {
 		return nil, nil, err
 	}
```
Comment (bhaney) [inline components/audioinput/client.go:52]: Does `Read` also need the extra parameters fed to it?

File: robot/web/web.go [hunk 2]
```diff
@@ -176,7 +176,7 @@ func allAudioSourcesToDisplay(theRobot robot.Robot) map[string]gostream.AudioSou
 			continue
 		}
 
-		sources[name] = input
+		sources[name] = audioinput.NewGostreamWrapper(input)
```
Comment (bhaney) [inline robot/web/web.go:179]: This essentially strips away the extra params before feeding AudioInput to the streamer. So if a user's custom AudioInput needed those extra params, streaming wont work on the app page.

**Review by edaniels — APPROVED** [review]

File: components/audioinput/client.go (hunk 1 above)
Comment (cheukt) [reply components/audioinput/client.go:52]: ya I think so

**Review by cheukt — CHANGES_REQUESTED** [review]
would be good to add some simple tests to make sure extra is getting passed through to the client and server like https://github.com/viamrobotics/rdk/pull/1545/files#diff-45b6db493935132fd510c7908cfbaa8d0ed3b0a6fe56e16281c692e17b8a5b8d and https://github.com/viamrobotics/rdk/pull/1545/files#diff-cb596d6d9d4b0bd2ad903c58b7c464690d348c2655e22b3eeff64b97acfdef00

File: robot/web/web.go (hunk 2 above)
Comment (cheukt) [reply robot/web/web.go:179]: can you add a comment here?

**Comment by bhaney** [comment]
Not gonna do this right now (and will probably do it differently in the future)

**Pull request closed without merging**